### PR TITLE
Separate piTest from piControl and use CMake instead of Make [REVPI-3198]

### PR DESCRIPTION
**Pull request by tboehler1**

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+install(FILES piTest.1 DESTINATION usr/share/man/man1)
```
Comment (hjokunbus) [inline doc/CMakeLists.txt:1]: Not part of this PR, but: Please keep in mind that this has to respect the `nodoc` option if it shoud ever be compatible with Debian policy version 4.0.0. Version 3.9.7 recommends a seperate package.
https://www.debian.org/doc/debian-policy/upgrading-checklist.html

File: piControlIf.c [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+/***********************************************************************************/
+/*!
+ * @brief Open Pi Control Interface
+ *
+ * Initialize the Pi Control Interface
+ *
+ ************************************************************************************/
+void piControlOpen(void)
+{
+        /* open handle if needed */
```
Comment (hjokunbus) [inline piControlIf.c]: Indention has been halved in regard to the original. Why?

**Comment by hjokunbus** [comment]
I see no major issues here so far, since other than whitespace and paths the C source hasn't being changed.

File: piControlIf.c (hunk 2 above)
Comment (tboehler1) [reply piControlIf.c]: @naruxde and me decided to try and keep a unified style in the source - this is also the reason the `.editorconfig` file was introduced.
If this is not an acceptable change I can revert it.

File: doc/CMakeLists.txt (hunk 1 above)
Comment (tboehler1) [reply doc/CMakeLists.txt:1]: I wasn't aware of this, thanks for the heads up!

File: piControlIf.c (hunk 2 above)
Comment (hjokunbus) [reply piControlIf.c]: I leave it to the Kernel developers to decide. They (who maintained the code in the past) followed the Linux coding style rules, and I think the 8 char identation is the first rule.
https://www.kernel.org/doc/html/v6.1/process/coding-style.html

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+set(TARGET piTest)
+set(SOURCES
+	piTest.c
+	piControlIf.c
+)
+
+add_executable(${TARGET} ${SOURCES})
+target_include_directories(${TARGET} PRIVATE ../inc ../lib/piControl)
+
+install(TARGETS ${TARGET} DESTINATION usr/bin)
```
Comment (iluminat23) [inline src/CMakeLists.txt:10]: Please use [GNUInstallDirs](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) for paths. See https://github.com/RevolutionPi/revpi-tools/blob/master/CMakeLists.txt 

I think `RUNTIME` needs to be added as well: https://cmake.org/cmake/help/latest/command/install.html#installing-targets

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# https://stackoverflow.com/a/41037224
+# TODO: this disregards any prefix set for installing
+install(CODE "execute_process( \
+	COMMAND ${CMAKE_COMMAND} -E create_symlink \
+	usr/bin/piControlReset usr/bin/${TARGET})")
```
Comment (iluminat23) [inline src/CMakeLists.txt:15]: This is not the correct symlink. The original symlink to relative so this symlink needs to be relative as well.
Also prefixing is mandatory. Don't install to usr.
Maybe it would be better to create the symlink in the build directory and install the symlink along the binary in the install step.

File: src/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+set(TARGET piTest)
+set(SOURCES
+	piTest.c
+	piControlIf.c
+)
+
+add_executable(${TARGET} ${SOURCES})
+target_include_directories(${TARGET} PRIVATE ../inc ../lib/piControl)
```
Comment (iluminat23) [inline src/CMakeLists.txt:8]: As I can't comment on the file rename I do it here. Please use include not inc as directory name.

**Review by iluminat23 — CHANGES_REQUESTED** [review]
Please add the URL and the commit id from the piControl repo when you got the piTest version. Also create a PR to remove pitest from picontrol. Then you can add piControl as submodule without pitest included in piControl.

File: src/CMakeLists.txt (hunk 4 above)
Comment (tboehler1) [reply src/CMakeLists.txt:15]: > Maybe it would be better to create the symlink in the build directory

I'm unsure how this works. It seems that, if I'm understanding this correctly, because `DESTINATION` is not given, `${CMAKE_INSTALL_PREFIX}` does not honor the `--prefix` flag in `cmake --install . --prefix $PREFIX` when using the `install` directive in lines 13-15. Just using `${CMAKE_INSTALL_BINDIR}/piControlReset` yields `bin/piControlReset`, not `$PREFIX/bin/piControl` like it does for the install target above.

If I'm building the file into the build directory, how would I refer to that file to install it? `install(FILES ...)` refers to files in the source directory, and hard coding the build directory is not an option.

**Comment by tboehler1** [comment]
The piControl submodule is now pinned to the latest commit, namely https://github.com/RevolutionPi/piControl/commit/8c74c7435c8d965c6ed915261d008c13692768c9, which removes piTest.

File: src/CMakeLists.txt (hunk 4 above)
Comment (iluminat23) [reply src/CMakeLists.txt]: The target of the symlink is allways `./piTest`. The symlink like the executeable are installed to the same directory.

So it should not a big issue to create a symlink in the build directory along with the executeable. And in the install step it should be possible to install the symlink like any other file (maybe).
